src/wxflow/hsi.py

```diff
@@ -0,0 +1,442 @@
+
+        opts : str | List
+                List or string of additional options to send to hsi.
+
+        Returns
+        -------
+        output : str
+                Concatenated output and error of the hsi put command.
+        """
+        arg_list = []
+
+        # Convert to str to handle Path objects
+        target = str(target)
+        source = str(source)
+
+        # Parse any hsi options
+        arg_list.extend(Hsi._split_opts(opts))
+
+        arg_list.append("put")
+        arg_list.append(source + " : " + target)
+        output = self._hsi(arg_list)
+
+        return output
+
+    def chmod(self, mod: str, target: str, hsi_opts: Union[List, str] = "",
+              chmod_opts: Union[List, str] = "") -> str:
+        """ Method to change the permissions of a file or directory on HPSS
+
+        Parameters
+        ----------
+        mod : str
+                Permissions to set for the file or directory,
+                e.g. "640", "o+r", etc.
+
+        target : str
+                Full path of the target location of the file on HPSS.
+
+        hsi_opts : list | str
+                Options to send to hsi.
+
+        chmod_opts : list | str
+                Options to send to chmod. See "hsi chmod -?" for more details.
+
+        Return
+        ------
+        output : str
+                Concatenated output and error of the hsi chmod command.
+        """
+
+        arg_list = []
+
+        # Parse any hsi options
+        arg_list.extend(Hsi._split_opts(hsi_opts))
+
+        arg_list.append("chmod")
+
+        # Parse any chmod options
+        arg_list.extend(Hsi._split_opts(chmod_opts))
+
+        arg_list.append(mod)
+        arg_list.append(target)
+        output = self._hsi(arg_list)
+
+        return output
+
+    def chgrp(self, group_name: str, target: str, hsi_opts: str = "",
+              chgrp_opts: str = "") -> str:
+        """ Method to change the group of a file or directory on HPSS
+
+        Parameters
+        ----------
+        group_name : str
+                The group to which ownership of the file/directory is to be set.
+
+        target : str
+                Full path of the target location of the file on HPSS.
+
+        hsi_opts : str
+                String of options to send to hsi.
+
+        chgrp_opts : str
+                Options to send to chgrp.  See "hsi chgrp -?" for more details.
+
+        Returns
+        -------
+        output : str
+                Concatenated output and error of the hsi chgrp command.
+        """
+
+        arg_list = []
+
+        # Parse any hsi options
+        arg_list.extend(Hsi._split_opts(hsi_opts))
+
+        arg_list.append("chgrp")
+
+        # Parse any chgrp options
+        arg_list.extend(Hsi._split_opts(chgrp_opts))
+
+        arg_list.append(group_name)
+        arg_list.append(target)
+        output = self._hsi(arg_list)
+
+        return output
+
+    def rm(self, target: str, hsi_opts: str = "", rm_opts: str = "") -> str:
```

Comment:
```suggestion
    def rm(self, target: str, recursive: bool = False, hsi_opts: str = "", rm_opts: str = "") -> str:
```
Instead of `-r`, provide an option `recursive`

Reply:
Done.